src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp

```diff
@@ -189,8 +189,51 @@ QList<Task*> PCRPrimerDesignForDNAAssemblyTask::onSubTaskFinished(Task* subTask)
 }
 
 QString PCRPrimerDesignForDNAAssemblyTask::generateReport() const {
-    //TODO - report
-    return QString();
+    QString report("<br>"
+                   "<br>"
+                   "<h3>%1</h3>");
+    SAFE_POINT(!sequence.isEmpty(), tr("Empty sequence"), report.arg(tr("Error, see log.")))
+    if (aForward.isEmpty() && aReverse.isEmpty() && b1Forward.isEmpty() && b1Reverse.isEmpty() && b2Forward.isEmpty() &&
+        b2Reverse.isEmpty() && b3Forward.isEmpty() && b3Reverse.isEmpty()) {
+        return report.arg(tr("There are no primers that meet the specified parameters."));
+    }
+
+    auto getPairReport = [&](U2Region forward, U2Region reverse, const QString &primerName) -> QString {
```

Comment:
I think, it's better to create a separate function - this is too much code for lambda